docs/backend/testing/running-integration-tests-with-routing-middleware-enabled.md

```diff
@@ -0,0 +1,105 @@
+
+The name of the file should correspond to the handler that the endpoint calls, `{handler}_test.go`. `pkg/handlers/internalapi/ppm_shipment.go` would have a corresponding test file in `pkg/handlers/routing/internalapi_test/ppm_shipment_test.go`.
+
+Generally, the test name and location should be
+```pkg/handlers/routing/{api_name}api_test/{handler_name}_test.go```
+
+## Test structure
+
+In general, here's how to set up the file created above.
+
+Include the test as part of the `{api_name}api_test` package, and any imports, as usual.
+
+For each request type, write a function whose receiver is the API's test suite. For example, tests of updating a pro-gear weight ticket in the internal API's suite would look like:
+
+```
+func (suite *InternalAPISuite) TestUpdateProgearWeightTicket() {
+  ...
+}
+```
+
+For each test, call `Run` on the suite, with a description of the test, and the test function.
+
+```
+  suite.Run("Unauthorized progear weight ticket update by another service member", func() { ... })
+```
+
+The body of the test function will likely contain some data set up -- using factories to build a pro-gear weight ticket and service member, for example -- which we can use to build the endpoint, body, and request, and get the response.
+### Endpoint
+
+Build the endpoint from the corresponding Swagger definition. For example, the `swagger-def/internal.yaml` defines the update path for pro-gear weight tickets as
+
+```
+/ppm-shipments/{ppmShipmentId}/pro-gear-weight-tickets/{proGearWeightTicketId}
+
+```
+
+Build the endpoint path, for example, assuming the `ppmShipment` and `proGearWeightTicket` are created,
+
+```
+endpointPath := fmt.Sprintf("internal/ppm-shipments/%s/pro-gear-weight-tickets/%s", ppmShipment.ID.String(), progearWeightTicket.ID.String())
+```
+
+### The request body
+
+Swagger generates types that define the shape a request body should take, and we can leverage those types in writing tests.
+
+For our example, our request body type is `internalmessages.UpdateProGearWeightTicket`, which Swagger generates for us in `pkg/gen/internalmessages/update_pro_gear_weight_ticket.go`.
+
+To build out the request body, we create an object of this type, marshal it to JSON, check for errors, then cast it to a buffer using `bytes.NewBuffer`.
+
+```
+body := &internalmessages.UpdateProGearWeightTicket{
+  Description:      handlers.FmtString("My cool stuff"),
+  HasWeightTickets: true,
+  Weight:           handlers.FmtInt64(4000),
+  BelongsToSelf:    true,
+}
+
+jsonBody, err := json.Marshal(body)
+
+suite.FatalNoError(err)
+
+bodyBuffer := bytes.NewBuffer(jsonBody)
+```
+### The request
+
+Now we can build the request from the endpoint and body. Since our request is hitting the customer side of the application, we'll use an authenticated MilMove request, and a `PATCH` method. We will also set the `Content-Type` header.
```

Comment:
**thought**: You mention an authenticated MilMove request here and show it in the example, which is great. I wonder if we should also mention the other possible request types like the unauthenticated version, or the office and prime versions:

https://github.com/transcom/mymove/blob/d491f8db417e9d3bb8711fffba47fb2aa0163b79/pkg/handlers/routing/base_routing_suite.go#L221-L307

What do you think?

Reply:
Yeah, that makes sense. I added reference to the other versions.